share/functions/__fish_print_help.fish

```diff
@@ -110,7 +110,7 @@ function __fish_print_help --description "Print help message for the specified f
     begin
         set -l pager less
         set -q PAGER
-        and set pager $PAGER
+        and echo $PAGER | read -a pager
```

Comment:
We prefer to use `eval set pager $PAGER` for these, probably because it allows to escape spaces in the path of `$PAGER`.
(Similar uses found via `git grep -E '\b(BROWSER|PAGER|EDITOR|VISUAL)\b' -- share/functions`)

Reply:
Once #6366 is merged these should probably be using `read --tokenize` instead.

Reply:
> We prefer to use `eval set pager $PAGER` for these, probably because it allows to escape spaces in the path of `$PAGER`.
> (Similar uses found via `git grep -E '\b(BROWSER|PAGER|EDITOR|VISUAL)\b' -- share/functions`)

Hmm, I'm concerned that this executes arbitrary command substitutions. Admittedly, we're about to run an arbitrary command from this env var *anyway*, but it means that this construct can't be used uncritically.

Reply:
#6366 is merged, so you can just change the `read` to `read -at pager`.

Reply:
Indeed, `read` avoids evaluating command substitutions is good.
`eval` matches bash's behavior more closely, the "features" we'd lose include expansion of unexpanded `~`, variables or quotes. Such values are probably rare in environment variables, so it seems fine to use read.